editor/editor_data.cpp

```diff
@@ -936,6 +936,7 @@ void EditorData::script_class_save_icon_paths() {
 
 	ProjectSettings::get_singleton()->set("_global_script_class_icons", d);
 	ProjectSettings::get_singleton()->save();
+	script_class_load_icon_paths();
```

Comment:
Moving this method here produces:
```
 core\io\image_loader.cpp:55 - Error opening file 'Null'.
 core\io\image_loader.cpp:55 - Error opening file 'Null'.
```
while opening a `class_name` script with no editor icon:

```
godot.windows.tools.64.exe!ImageLoader::load_image(String p_file, Ref<Image> p_image, FileAccess * p_custom, bool p_force_linear, float p_scale) Line 55 (d:\src\godot\core\io\image_loader.cpp:55)
godot.windows.tools.64.exe!EditorNode::_load_custom_class_icon(const String & p_path) Line 3704 (d:\src\godot\editor\editor_node.cpp:3704)
godot.windows.tools.64.exe!EditorNode::get_object_icon(const Object * p_object, const String & p_fallback) Line 3728 (d:\src\godot\editor\editor_node.cpp:3728)
godot.windows.tools.64.exe!EditorPath::update_path() Line 95 (d:\src\godot\editor\editor_path.cpp:95)
godot.windows.tools.64.exe!InspectorDock::update(Object * p_object) Line 394 (d:\src\godot\editor\inspector_dock.cpp:394)
godot.windows.tools.64.exe!EditorNode::_edit_current() Line 1884 (d:\src\godot\editor\editor_node.cpp:1884)
godot.windows.tools.64.exe!EditorNode::push_item(Object * p_object, const String & p_property, bool p_inspector_only) Line 1810 (d:\src\godot\editor\editor_node.cpp:1810)
godot.windows.tools.64.exe!InspectorDock::_resource_selected(const Ref<Resource> & p_res, const String & p_property) Line 289 (d:\src\godot\editor\inspector_dock.cpp:289)
godot.windows.tools.64.exe!InspectorDock::edit_resource(const Ref<Resource> & p_resource) Line 365 (d:\src\godot\editor\inspector_dock.cpp:365)
godot.windows.tools.64.exe!EditorNode::load_resource(const String & p_resource, bool p_ignore_broken_deps) Line 900 (d:\src\godot\editor\editor_node.cpp:900)
godot.windows.tools.64.exe!FileSystemDock::_select_file(const String & p_path, bool p_select_in_favorites) Line 850 (d:\src\godot\editor\filesystem_dock.cpp:850)
godot.windows.tools.64.exe!FileSystemDock::_tree_activate_file() Line 864 (d:\src\godot\editor\filesystem_dock.cpp:864)
godot.windows.tools.64.exe!call_with_variant_args_helper<FileSystemDock>(FileSystemDock * p_instance, void(FileSystemDock::*)() p_method, const Variant * * p_args, Callable::CallError & r_error, IndexSequence<> __formal) Line 136 (d:\src\godot\core\callable_method_pointer.h:136)
godot.windows.tools.64.exe!call_with_variant_args<FileSystemDock>(FileSystemDock * p_instance, void(FileSystemDock::*)() p_method, const Variant * * p_args, int p_argcount, Callable::CallError & r_error) Line 158 (d:\src\godot\core\callable_method_pointer.h:158)
godot.windows.tools.64.exe!CallableCustomMethodPointer<FileSystemDock>::call(const Variant * * p_arguments, int p_argcount, Variant & r_return_value, Callable::CallError & r_call_error) Line 185 (d:\src\godot\core\callable_method_pointer.h:185)
godot.windows.tools.64.exe!Callable::call(const Variant * * p_arguments, int p_argcount, Variant & r_return_value, Callable::CallError & r_call_error) Line 50 (d:\src\godot\core\callable.cpp:50)
godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant * * p_args, int p_argcount) Line 1077 (d:\src\godot\core\object.cpp:1077)
godot.windows.tools.64.exe!Object::emit_signal(const StringName & p_name, const Variant & p_arg1, const Variant & p_arg2, const Variant & p_arg3, const Variant & p_arg4, const Variant & p_arg5) Line 1132 (d:\src\godot\core\object.cpp:1132)
godot.windows.tools.64.exe!Tree::_gui_input(Ref<InputEvent> p_event) Line 2608 (d:\src\godot\scene\gui\tree.cpp:2608)
godot.windows.tools.64.exe!MethodBind1<Tree,Ref<InputEvent> >::call(Object * p_object, const Variant * * p_args, int p_arg_count, Callable::CallError & r_error) Line 867 (d:\src\godot\core\method_bind.gen.inc:867)
```

This change likely makes sense but doing so populates "Null" icon paths for each script with no icon. I'd leave that for another PR in fact. 🙂

Reply:
Sounds good. I'll revert this change for now.